litestar/contrib/sqlalchemy/repository/_async.py

```diff
@@ -300,9 +331,9 @@ async def get_one_or_none(
             The retrieved instance or None
         """
         with wrap_sqlalchemy_exception():
-            statement = statement if statement is not None else self.statement
+            statement = self._to_lambda_stmt(statement)
             statement = self._filter_select_by_kwargs(statement=statement, kwargs=kwargs)
-            instance = (await self._execute(statement)).scalar_one_or_none()
+            instance = cast("Result[tuple[ModelT]]", (await self._execute(statement))).scalar_one_or_none()
```

Comment:
Lambda statements don't support proper type inference?

Reply:
I'll do some additional testing on this before we merge it.  I want to say it doesn't, but I tried quite a few things and I now am a little unsure :)

Reply:
So, we can't do the type inference the same way we do for a `select`.  It raises an exception when it's hinted the same way.

Reply:
Sad. Could you find out if SQLA intends to support this at some point?